### PR TITLE
K8SPG-38 Cluster self-healing test

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/faa88ca816631388f8080458e1febed84b34e7a3
image: `perconalab/percona-postgresql-operator:PR-81-faa88ca8-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-faa88ca8-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-faa88ca8-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-faa88ca8-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-faa88ca8-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-faa88ca8-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| demand-backup | failed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/07bc80e67240b2c90909fdc283c4c135fcc9ef5a
image: `perconalab/percona-postgresql-operator:PR-81-07bc80e6-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-07bc80e6-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-07bc80e6-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-07bc80e6-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-07bc80e6-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-07bc80e6-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| data-migration-gcs | passed |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/fdc84ac27c949af6bce6a9eb3dfd740476b7e28e
image: `perconalab/percona-postgresql-operator:PR-81-fdc84ac2-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-fdc84ac2-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-fdc84ac2-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-fdc84ac2-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-fdc84ac2-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-fdc84ac2-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/6161d2ec591b178b8046103dbfc250d22d012113
image: `perconalab/percona-postgresql-operator:PR-81-6161d2ec-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-6161d2ec-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-6161d2ec-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-6161d2ec-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-6161d2ec-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-6161d2ec-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/d825a9ad25837780e1e667253bb2db92aa227ebe
image: `perconalab/percona-postgresql-operator:PR-81-d825a9ad-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-d825a9ad-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-d825a9ad-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-d825a9ad-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-d825a9ad-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-d825a9ad-pgo-deployer`

File: e2e-tests/conf/chaos-network-loss.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: chaos-mesh.org/v1alpha1
+kind: NetworkChaos
+metadata:
+  name: network-loss-example
+  namespace: namespace
+spec:
+  action: loss
+  mode: one
+  loss:
+    loss: "25"
```
Comment (hors) [inline e2e-tests/conf/chaos-network-loss.yml:10]: why only 25? please use all confs from pxc e.g. https://github.com/percona/percona-xtradb-cluster-operator/blob/main/e2e-tests/conf/chaos-network-loss.yml

**Review by hors — CHANGES_REQUESTED** [review]

File: e2e-tests/conf/chaos-network-loss.yml (hunk 1 above)
Comment (cap1984) [reply e2e-tests/conf/chaos-network-loss.yml:10]: We are using chaos-mesh version 2, pxc uses old chaos-mesh version 0.5.1

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | failed |
| init-deploy | failed |
| scaling | failed |
| recreate | failed |
| affinity | failed |
| monitoring | failed |
| self-healing | failed |
| data-migration-gcs | failed |
| clone-cluster | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/8ed92829eb56ee6e049c7904a7d85fd1925c1c0b
image: `perconalab/percona-postgresql-operator:PR-81-8ed92829-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-8ed92829-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-8ed92829-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-8ed92829-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-8ed92829-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-8ed92829-pgo-deployer`

File: e2e-tests/conf/chaos-network-loss.yml (hunk 1 above)
Comment (cap1984) [reply e2e-tests/conf/chaos-network-loss.yml]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | passed |
| init-deploy | passed |
| data-migration-gcs | passed |
| scaling | passed |
| recreate | passed |
| affinity | passed |
| monitoring | passed |
| self-healing | passed |
| clone-cluster | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/81/commits/0418903b9a0e28059262f6a9c789e90fcb2ca562
image: `perconalab/percona-postgresql-operator:PR-81-0418903b-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-81-0418903b-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-81-0418903b-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-81-0418903b-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-81-0418903b-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-81-0418903b-pgo-deployer`

**Pull request merged**